data_host/chrome_extension/src/pages/Popup/Popup.jsx

```diff
@@ -27,18 +26,86 @@ const renderWebpage = (hostname, paths, key) => {
   );
 };
 
+const renderWebpages = (webpages) =>
+  Object.entries(webpages).map(([hostname, paths], i) =>
+    renderWebpage(hostname, paths, i)
+  );
+
+const renderFractalData = ({ id }) => (
+  <ul>
+    <li key="fractal-id">
+      <strong>ID:</strong> {id || 'No ID set'}
+    </li>
+  </ul>
+);
+
+const onSubmit = async (dispatch, { fractalID }, form) => {
+  await dispatch(addFractalID(fractalID));
+  form.reset();
+};
+
+const validate = (data) => {
+  switch (true) {
+    case Object.keys(data).length === 0:
+      return {};
+    case !data.fractalID.startsWith('0x'):
+      return { fractalID: 'Must be a valid message' };
```

Comment:
This is one my big pet peeves. Anything that says "do it right" without saying what "right" is.

Instead, say something like "Expected hex-encoded byte sequence beginning with '0x'".

Reply:
I agree but honestly, I didn't give it much thought since it was just an example validation to ensure it would work. Ideally, I'd check Fractal's signature here. I'll replace the text.